File: client/src/app/interfaces/chat-message.ts
import { MessageType } from '@common/interfaces/message-type';

export interface ChatMessage {
    id: number;
    timestamp: Date;
    username: string;
    message: string;
    roomId?: string;         
    type?: MessageType;   
    senderUid?: string;   
    avatarURL?: string | null; 
}

File: client-leger/lib/models/chat_message.dart
class ChatMessage {
  final String id;
  final String roomId;
  final String username;
  final String message;
  final DateTime timestamp;
  final String type;
  final String? avatarURL;

  const ChatMessage({
    required this.id,
    required this.roomId,
    required this.username,
    required this.message,
    required this.type,
    required this.timestamp,
    this.avatarURL,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> j) => ChatMessage(
    id: (j['_id'] ?? '').toString(),
    roomId: (j['roomId'] ?? '').toString(),
    username: (j['username'] ?? '').toString(),
    message: (j['message'] ?? '').toString(),
    type: (j['type'] ?? 'global').toString(),
    timestamp: DateTime.parse(j['timestamp'] ?? "-"),
    avatarURL: j['avatarURL']?.toString(),
  );
}
